internal/local/structure_v2.go

```diff
@@ -536,6 +560,13 @@ func writeDataSources(rootDir string, dataSources []DataSourceStructure) error {
 			); err != nil {
 				return err
 			}
+			if err := WriteFile(
+				filepath.Join(dir, name, fmt.Sprintf("%s", rule["database"]), fmt.Sprintf("%s", rule["collection"]), FileRelationships.String()),
```

Comment:
I didn't notice these before, but we can avoid using `fmt.Sprintf("%s"...` to cast to strings (especially for known fields `database` and `collection` here).

considering we need db and coll for 3 different filepaths here, lets DRY this up a bit:

```go
var database, collection string
if db, ok := rule["database"]; ok {
  if db, ok := db.(string); ok { database = db }
}
// same for collection
ruleDir := filepath.Join(dir, name, database, collection)
```